engine/collection/message_hub/message_hub.go

```diff
@@ -241,10 +241,14 @@ func (h *MessageHub) processQueuedMessages(ctx context.Context) error {
 	}
 }
 
-// processQueuedTimeout performs actual processing of model.TimeoutObject, as a result of successful invocation
-// broadcasts timeout object to consensus committee.
+// processQueuedTimeout propagates the timeout to the collector cluster:
+//   - the node's internal `timeoutAggregator`
+//   - broadcast to all other cluster participants (excluding myself)
+//
 // No errors are expected during normal operations.
 func (h *MessageHub) processQueuedTimeout(timeout *model.TimeoutObject) error {
```

Comment:
I find that there is a subtle design asymmetry for processing votes, proposals and timeouts:
* receiving such a message from a _different node_, we _directly_ forward the message to the `compliance.Engine`, `VoteAggregator` or `TimeoutAggregator` (-> [code](https://github.com/onflow/flow-go/blob/de7fbb380ffc10ded1db53626f7e329dfe789955/engine/collection/message_hub/message_hub.go#L433-L452))
* In comparison, for our own messages, we queue the messages first and then lump together the sending to _external nodes_ as well as forwarding the message to the _internal_ components.

To me, it would be more intuitive to have a symmetric implementation:
* as part of `OnOwnProposal`, `OnOwnTimeout` and `OnOwnVote`, we forward the message to the internal component _right away_. There is no need send the messages to the queue, because the internal components are non-blocking.
* the methods `processQueuedProposal`, `processQueuedTimeout`, `processQueuedVote` and `processQueuedMessages` are only for sending the messages to _other_ peers. We could rename them to `sendOwnProposal`,  `sendOwnTimeout`, `endOwnVote` and `sendOwnMessages`.

To me, this would make the logic more consistent.